modelservice/src/main/java/com/expedia/adaptivealerting/modelservice/dao/es/ElasticSearchDetectorMappingService.java

```diff
@@ -0,0 +1,357 @@
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Timer;
+import com.expedia.adaptivealerting.modelservice.dao.DetectorMappingService;
+import com.expedia.adaptivealerting.modelservice.model.CreateDetectorMappingRequest;
+import com.expedia.adaptivealerting.modelservice.model.Detector;
+import com.expedia.adaptivealerting.modelservice.model.DetectorMapping;
+import com.expedia.adaptivealerting.modelservice.model.DetectorMatchResponse;
+import com.expedia.adaptivealerting.modelservice.model.MatchingDetectorsResponse;
+import com.expedia.adaptivealerting.modelservice.model.SearchMappingsRequest;
+import com.expedia.adaptivealerting.modelservice.util.QueryUtil;
+import com.google.gson.Gson;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.lucene.search.join.ScoreMode;
+import org.elasticsearch.action.admin.indices.mapping.get.GetMappingsRequest;
+import org.elasticsearch.action.admin.indices.mapping.get.GetMappingsResponse;
+import org.elasticsearch.action.admin.indices.mapping.put.PutMappingRequest;
+import org.elasticsearch.action.delete.DeleteRequest;
+import org.elasticsearch.action.get.GetRequest;
+import org.elasticsearch.action.get.GetResponse;
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.action.index.IndexResponse;
+import org.elasticsearch.action.search.SearchRequest;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.document.DocumentField;
+import org.elasticsearch.common.unit.TimeValue;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.index.query.BoolQueryBuilder;
+import org.elasticsearch.index.query.ConstantScoreQueryBuilder;
+import org.elasticsearch.index.query.NestedQueryBuilder;
+import org.elasticsearch.index.query.QueryBuilders;
+import org.elasticsearch.index.query.RangeQueryBuilder;
+import org.elasticsearch.percolator.PercolateQueryBuilder;
+import org.elasticsearch.search.SearchHit;
+import org.elasticsearch.search.SearchHits;
+import org.elasticsearch.search.builder.SearchSourceBuilder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import static com.expedia.adaptivealerting.modelservice.dao.es.DetectorMappingEntity.LAST_MOD_TIME_KEYWORD;
+
+@Component
+@Slf4j
+@SuppressWarnings({"PMD.ExcessiveImports", "PMD.AvoidThrowingRawExceptionTypes"})
+public class ElasticSearchDetectorMappingService implements DetectorMappingService {
```

Comment:
Have we looked into Spring Data Elasticsearch? I would expect Spring Data to eliminate this sort of code.

Reply:
This is noted as a separate change.